### PR TITLE
shellcheck: Make all bash scripts pass shellcheck

### PR DESCRIPTION
**Pull request by mestery**
This commit adds a "shellcheck" Makefile target, and fixes all of
the bash scripts to pass a shellcheck [1] run. Shellcheck does
static analysis of bash scripts. In the past, it's saved me hours
of debugging things in production, I find it useful to use early
in a projects lifetime. Once we have proper CI/CD for IPDK, we'd
want to run the "shellcheck" target on each pull request before
merging, for example.

[1] https://github.com/koalaman/shellcheck

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: build/IPDK_Container/scripts/build_p4sde.sh [hunk 1]
```diff
@@ -19,11 +20,11 @@ git config --global url."https://github.com/".insteadOf git@github.com:
 git config --global url."https://".insteadOf git://
 
 WORKDIR=$1
-cd ${WORKDIR}
+cd "${WORKDIR}" || exit
 
 echo "Removing p4-sde directory if it already exists"
 if [ -d "p4-sde" ]; then rm -Rf p4-sde; fi
-mkdir $1/p4-sde && cd $1/p4-sde
+mkdir "$1"/p4-sde && cd "$1"/p4-sde || exit
```
Comment (vsureshkumarp) [inline build/IPDK_Container/scripts/build_p4sde.sh:27]: File level error flag "set -e" enabled i.e. exit if any command returns non-zero exit status.
Is there any specific reason for adding command level exit.

File: build/IPDK_Container/scripts/set_hugepages.sh [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+mkdir -p /mnt/huge
 mount -t hugetlbfs nodev /mnt/huge
 echo -e "nodev /mnt/huge hugetlbfs\n" >> /etc/fstab
 
 echo "vm.nr_hugepages = 4096" >> /etc/sysctl.conf
 #sysctl -p /etc/sysctl.conf
 
-echo 1024 | sudo tee /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
-echo 1024 | sudo tee /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
+if [ ! -d /sys/devices/system/node/node0/hugepages-2048kB ] ; then
```
Comment (vsureshkumarp) [inline build/IPDK_Container/scripts/set_hugepages.sh:13]: Can you please add a comment here if possible?  Additional comments can provide some context on this change.

File: build/IPDK_Container/scripts/set_hugepages.sh [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+	echo 1024 | tee /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
+else
+	echo 1024 | sudo tee /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
+	echo 1024 | sudo tee /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
```
Comment (vsureshkumarp) [inline build/IPDK_Container/scripts/set_hugepages.sh:17]: If possible, can you please add an additional check to verify NUMA node exists before running the hugepages configuration on it.  As you are already making changes in the same area, requesting you to take care of it if possible.

File: build/IPDK_Container/scripts/set_hugepages.sh (hunk 2 above)
Comment (mestery) [reply build/IPDK_Container/scripts/set_hugepages.sh:13]: Yes, I'll add one here.

File: build/IPDK_Container/scripts/build_p4sde.sh (hunk 1 above)
Comment (mestery) [reply build/IPDK_Container/scripts/build_p4sde.sh:27]: Spellcheck requires this.

File: build/IPDK_Container/scripts/set_hugepages.sh [hunk 4]
```diff
@@ -1,16 +1,27 @@
...
 
 echo "vm.nr_hugepages = 4096" >> /etc/sysctl.conf
 #sysctl -p /etc/sysctl.conf
 
-echo 1024 | sudo tee /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
-echo 1024 | sudo tee /sys/devices/system/node/node1/hugepages/hugepages-2048kB/nr_hugepages
+#
+# Check if the node version of hugepages exists, and set hugepages if so.
+#
+if [ -d /sys/devices/system/node/node0/hugepages-2048kB ] ; then
```
Comment (vsureshkumarp) [inline build/IPDK_Container/scripts/set_hugepages.sh:16]: Need to correct the folder path. Correct path is 
/sys/devices/system/node/node0/hugepages/hugepages-2048kB

Also requires additional change for checking NUMA node1 existence too before running the configuration settings.
 /sys/devices/system/node/node0/hugepages/hugepages-2048kB
---
 /sys/devices/system/node/node1/hugepages/hugepages-2048kB

Comment (vsureshkumarp) [reply build/IPDK_Container/scripts/set_hugepages.sh:16]: Somehow some part of the comment text using different formatting options.

Comment (mestery) [reply build/IPDK_Container/scripts/set_hugepages.sh]: Corrected the path.

**Comment by mestery** [comment]
This works now and I am able to get a build to pass with all the shellcheck changes included.

**Pull request merged**